app/controllers/blogs_controller.rb

```diff
@@ -0,0 +1,74 @@
+class BlogsController < ApplicationController
+  before_action :set_blog, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
```

Comment:
You need to create a custom `before_action` method to restrict users to specific actions and that custom method should be declared in `ApplicationController` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)).

```ruby
before_action :signed_in_only!, only: [:new, :create]

def signed_in_only!
  # current_user in ApplicationController must be defined as helper method to make it work
  redirect_to  new_user_session_path unless current_user
end
```